src/login/logind-session.c

```diff
@@ -95,10 +93,10 @@ Session* session_free(Session *s) {
 
         s->timer_event_source = sd_event_source_unref(s->timer_event_source);
 
-        session_drop_controller(s);
+        session_drop_controller(s, drop_resources);
 
-        while ((sd = hashmap_first(s->devices)))
-                session_device_free(sd, true);
+        for (SessionDevice *sd; (sd = hashmap_first(s->devices)); )
```

Comment:
not a fan of misuing a for() loop as a weird while() loop... makes stuff less readable not more readable